src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+                retval, iconf->iface);
+        SCReturnInt(retval);
+    }
+
+    if (!rte_eth_dev_is_valid_port(iconf->port_id)) {
+        SCLogError(SC_ERR_DPDK_INIT, "Specified port %d is invalid", iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = rte_eth_dev_socket_id(iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) invalid socket id (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    } else {
+        iconf->socket_id = retval;
+    }
+
+    retval = rte_eth_dev_info_get(iconf->port_id, &dev_info);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during getting device info (port %" PRIu16 ")",
+                retval, iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    if (iconf->nb_rx_queues > dev_info.max_rx_queues) {
+        errno = ERANGE;
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Number of configured RX queues of port %" PRIu16
+                " is higher than maximum allowed (%" PRIu16 ")",
+                iconf->port_id, dev_info.max_rx_queues);
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->nb_tx_queues > dev_info.max_tx_queues) {
+        errno = ERANGE;
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Number of configured TX queues of port %" PRIu16
+                " is higher than maximum allowed (%" PRIu16 ")",
+                iconf->port_id, dev_info.max_tx_queues);
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->mtu > dev_info.max_mtu || iconf->mtu < dev_info.min_mtu) {
+        errno = ERANGE;
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Loaded MTU of port %" PRIu16 " is out of bounds. "
+                "Min MTU: %" PRIu16 " Max MTU: %" PRIu16,
+                iconf->port_id, dev_info.min_mtu, dev_info.max_mtu);
+        SCReturnInt(-errno);
+    }
+
+    DeviceInitPortConf(iconf, &dev_info, &port_conf);
+    if (port_conf.rxmode.offloads & DEV_RX_OFFLOAD_CHECKSUM) {
+        // Suricata does not need recalc checksums now
+        iconf->checksum_mode = CHECKSUM_VALIDATION_DISABLE;
+    }
+
+    retval = rte_eth_dev_configure(
+            iconf->port_id, iconf->nb_rx_queues, iconf->nb_tx_queues, &port_conf);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during configuring the device (port %u)",
+                retval, iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = rte_eth_dev_adjust_nb_rx_tx_desc(
+            iconf->port_id, &iconf->nb_rx_desc, &iconf->nb_tx_desc);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Error (err=%d) during adjustment of device queues descriptors (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = iconf->flags & DPDK_MULTICAST ? rte_eth_allmulticast_enable(iconf->port_id)
```

Comment:
I assume you are configuring for MAC multicast and not IP multicast. Can you please cross check if suricata multicast talks about IP multicast or not?